lib/archethic/self_repair/network_chain.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Archethic.SelfRepair.NetworkChain do
+  @moduledoc """
+  Synchronization of one or multiple network chains.
+  Asynchronous functions use a NetworkChainWorker to avoid concurrent runs.
+  """
+  alias Archethic.Crypto
+  alias Archethic.OracleChain
+  alias Archethic.P2P
+  alias Archethic.Reward
+  alias Archethic.SelfRepair
+  alias Archethic.SelfRepair.NetworkChainWorker
+  alias Archethic.SharedSecrets
+  alias Archethic.TransactionChain
+
+  @doc """
+  Synchronize the network chain of given type.
+  It runs in a worker. Skips if worker is already running.
+  """
+  @spec asynchronous_resync(NetworkChainWorker.type()) :: :ok
+  def asynchronous_resync(network_chain_type) do
+    NetworkChainWorker.resync(network_chain_type)
+  end
+
+  @doc """
+  Synchronize the network chains of given types.
+  Every sync is done in its own worker, skip those already running.
+  """
+  @spec asynchronous_resync_many(list(NetworkChainWorker.type())) :: :ok
+  def asynchronous_resync_many(network_chain_types) do
+    Enum.each(
+      network_chain_types,
+      &asynchronous_resync(&1)
+    )
+  end
+
+  @doc """
+  Synchronize the network chain of given type.
+  Blocks the caller.
+  """
+  @spec synchronous_resync(NetworkChainWorker.type()) :: :ok | {:error, :network_issue}
+  def synchronous_resync(:node) do
+    :telemetry.execute([:archethic, :self_repair, :resync], %{count: 1}, %{network_chain: :node})
+
+    case P2P.fetch_nodes_list() do
+      {:ok, nodes} ->
+        nodes_to_resync = Enum.filter(nodes, &node_require_resync?/1)
+
+        # Load the latest node transactions
+        Task.Supervisor.async_stream_nolink(
+          Archethic.TaskSupervisor,
+          nodes_to_resync,
+          fn node ->
+            SelfRepair.resync(
+              Crypto.derive_address(node.first_public_key),
+              Crypto.derive_address(node.last_public_key)
```

Comment:
Here you should use `node.last_address` because deriving the last public key will return you the previous last address as the real last address is the hash of the next public key which we cannot know